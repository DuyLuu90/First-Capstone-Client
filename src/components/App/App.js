import React,{Component} from 'react';
import { Route, Switch} from 'react-router-dom'
import './App.css';

import TokenService from '../../services/token-service'
import {GeneralApiServices} from '../../services/api-service'

//IMPORT CONTEXT
import AppContext from '../../contexts/AppContext'

//IMPORT COMPONENTS:
import Header from '../Header/Header'
import LoginForm from '../Forms/LoginForm'
import RegistrationForm from '../Forms/RegistrationForm'
import ListItem from '../ListItem/ListItem'
//INPORT ROUTES:
import DemoPage from '../../routes/DemoPage/DemoPage'
import HomePage from '../../routes/HomePage/HomePage'
import AdminPage from '../../routes/AdminPage/AdminPage'
import MoviePage from '../../routes/MoviePage/MoviePage'
import UserPage from '../../routes/Profiles/UserPage'
import ArtistPage from '../../routes/Profiles/ArtistPage'
import NotFoundPage from '../../routes/NotFoundPage/NotFoundPage'

class App extends Component {
  state= {
    hasError: false,
    hasAuthToken: TokenService.hasAuthToken(),
    userid:'',
    isAdmin: false,
    first_name:''
  }

  componentDidMount(){
    const authToken= TokenService.getAuthToken()
    if (authToken) this.handleLoginSuccess()
  }

  handleLoginSuccess=()=>{
    const authToken= TokenService.getAuthToken()
    const userid= TokenService.parseJwt(authToken).userid
    GeneralApiServices.getItemById('users',userid)
        .then(user=>this.setState({
            hasAuthToken: TokenService.hasAuthToken(),
            userid: userid,
            isAdmin: user.admin,
            first_name: user.first_name
        }))
  }
  handleLogoutSuccess=()=>{
    TokenService.clearAuthToken()
    this.setState({
      hasAuthToken: TokenService.hasAuthToken(),
      isAdmin: false,
      userid: '',
      last_name: ''})
  }
  renderAdminPage=(props)=>{
    const Admin= (this.state.isAdmin)? <AdminPage {...props}/>:<NotFoundPage/>
    return Admin
  }

  render() {
    const {state:{hasAuthToken,userid,isAdmin,first_name}, 
    handleLoginSuccess,handleLogoutSuccess}= this
    const values= {
      hasAuthToken: hasAuthToken,userid:userid,
      isAdmin: isAdmin,first_name: first_name,
      handleLoginSuccess: handleLoginSuccess,
      handleLogoutSuccess: handleLogoutSuccess
    }
    return (
      <AppContext.Provider value={values}>
          <div className="App">
            <header className='App_header'>
              <Header token={this.state} onLogoutSuccess={this.handleLogoutSuccess}/>
            </header>
          <main className='App_main'>
            {this.state.hasError && <p>There was an error! Sorry for the inconvenience!</p>}
            <Switch>
              <Route exact path={'/'} component={HomePage}/>
              <Route path={'/demo'} component={DemoPage}/>
              <Route path={'/movies/genres/:genres'} component={ListItem}/>
              <Route path={'/movies/country/:country'} component={ListItem}/>
              <Route path={'/admin'} component={(props)=>this.renderAdminPage(props)}/>
              <Route path={'/register'} component={(props)=>
                  <RegistrationForm {...props} handleCancel={()=>props.history.goBack()}/>}/>
              <Route path={'/login'} component={(props)=>
                  <LoginForm {...props} onLoginSuccess={this.handleLoginSuccess}/>}
              />
              <Route path={'/movies/:id'} component={(props)=>
                  <MoviePage {...props} hasAuthToken={this.state.hasAuthToken}/>}
              />
              <Route path={'/users/:id'} component={(props)=>
                  <UserPage {...props} currentUserid={this.state.userid}/>}
              />
              <Route path={'/artists/:id'} component={ArtistPage}/>
              <Route component={NotFoundPage}/>
            </Switch>
          </main>
        </div>
      </AppContext.Provider>
    );
  }
}

export default App;