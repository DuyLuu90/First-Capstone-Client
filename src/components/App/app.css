.App{
    position: relative;
    min-height: 100vh;
  }

.App_main {
    margin-top: 70px;
    padding-bottom: 70px;
}

.App_main h1{
    text-align: center;
    padding: 10px;
    margin-bottom: 0;
    color: rgb(168, 153, 63);
}