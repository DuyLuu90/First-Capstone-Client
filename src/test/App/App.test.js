import React from 'react'
import ReactDOM from 'react-dom'
import App from '../../components/App/App'

import renderer from 'react-test-renderer'
import {BrowserRouter} from 'react-router-dom'


describe.only(`App Component`, () => {
    it ('Smoke Test-Render without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(
        <BrowserRouter>
            <App />
        </BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    });

    it('Snapshot test-Prevent unexpected change', () => {
//render the component and create a human-readable JSON file
        const myRenderedElement= renderer.create(
        <BrowserRouter>
            <App />
        </BrowserRouter>).toJSON();
//compare the rendered component to a saved version of the component 
        expect(myRenderedElement).toMatchSnapshot();
    })

})
