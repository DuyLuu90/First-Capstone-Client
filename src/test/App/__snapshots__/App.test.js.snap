// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Snapshot test-Prevent unexpected change 1`] = `
<div
  className="App"
>
  <header
    className="App_header"
  >
    <nav
      className="app_nav"
    >
      <h1>
        <a
          aria-label="home-page"
          href="/home"
          onClick={[Function]}
        >
           
           
          DramaPEDIA
        </a>
      </h1>
      <div
        className="menu"
      >
        <div
          className="Header-not-logged-in"
          onClick={[Function]}
        >
          <a
            aria-label="login-page"
            href="/login"
            onClick={[Function]}
          >
            Log in 
          </a>
          <a
            aria-label="registration-page"
            href="/register"
            onClick={[Function]}
          >
             Sign up
          </a>
          <a
            aria-label="demo-page"
            href="/"
            onClick={[Function]}
          >
            Welcome 
          </a>
        </div>
      </div>
      <div
        className="dropdown"
      />
    </nav>
  </header>
  <main
    className="App_main"
  >
    <div
      className="demo"
    >
      <nav>
        <span
          className="highlight"
          onClick={[Function]}
        >
          App tour
        </span>
        <span
          className=""
          onClick={[Function]}
        >
          Demo accounts
        </span>
      </nav>
      <div>
        <div
          className="tour_nav"
        >
          <span
            className="active"
            onClick={[Function]}
          >
            About
          </span>
          <span
            className=""
            onClick={[Function]}
          >
            Visitor
          </span>
          <span
            className=""
            onClick={[Function]}
          >
            User
          </span>
          <span
            className=""
            onClick={[Function]}
          >
            Admin
          </span>
        </div>
        <div
          className="tour_content"
        >
          <h2>
            About this app
          </h2>
          <div
            className="app_features"
          >
            <div
              className="text"
            >
              Welcome to 
              <strong>
                dramaPEDIA
              </strong>
              . Here you can see a variety of Films/TV Series from many countries around the world. You can create a new account and have your own personal page where you can save your favorite movies. To learn more about app features, please click on some of the tabs above.
            </div>
            <div
              className="featureImg"
            >
              <img
                alt="world cinema"
                src="/assest/worldCinema.PNG"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
  <footer
    className="app_footer"
  >
    <div>
      <span>
         
        Duy Luu 2020
      </span>
    </div>
    <a
      aria-label="email"
      href="mailto:duyluu90@hotmail.com"
      rel="noopener noreferrer"
      target="_blank"
    >
       
       
      EMAIL ME
    </a>
  </footer>
</div>
`;
